### PR TITLE
Change default astropy_helpers_version based on python2 support

### LINKED ISSUE #303: branches of astropy-helpers in cookiecoutter.json
@bsipocz mentioned in #286 about astropy-helpers in cookiecutter.json:

We may want to branch this to be v3.0 or 2.0.4 depending whether the users package is python3 only or not.

### PR DESCRIPTION
**Pull request by bsipocz**
apparently cookiecutter understands jinja2 templates, so we can have conditionals in the json without too much extra hacking.

Close #303

### CODE REVIEW

**Comment by bsipocz** [comment]
@Cadair - Minimum python was visioned to be optional, but if think it's better to keep it mandatory I'm happy to reuse it's content rather than asking a new option.

**Comment by Cadair** [comment]
You could do something like asking support py2 first and if you say no setting the default python 3 to 3.5 or something? I.e. modifying the default minimum python version.

**Comment by bsipocz** [comment]
Oh ok, I can definitely do that.

File: cookiecutter.json [hunk 1]
```diff
@@ -18,11 +18,12 @@
...
-    "astropy_helpers_version": "v2.0.2",
     "_parent_project": "astropy",
     "_install_requires": "astropy",
     "_copy_without_render": [
         "docs/_templates"
     ],
-    "__minimum_python_version__": ""
+    "support_python2": "n",
+    "astropy_helpers_version": "{% if cookiecutter.support_python2 == 'y' %}v2.0.6{% else %}v3.0.1{% endif %}",
+    "minimum_python_version": "{% if cookiecutter.support_python2 == 'n'%}3.5{% endif %}"
```
Comment (Cadair) [inline cookiecutter.json]: I think this needs to be `"3.5"` not `3.5` to fix the CI.

Comment (pllim) [reply cookiecutter.json]: Ah, yes, it is designed to be a string (to be compatible with non-cookie code in core and master template).

**Pull request merged**